src/prop/theory_preregistrar.cpp

```diff
@@ -39,13 +67,46 @@ void TheoryPreregistrar::addAssertion(TNode n, TNode skolem, bool isLemma) {}
 
 void TheoryPreregistrar::notifyActiveSkolemDefs(std::vector<TNode>& defs) {}
 
-void TheoryPreregistrar::notifySatLiteral(TNode n)
+void TheoryPreregistrar::notifySatLiteral(TNode n, bool cache)
 {
   // if eager policy, send immediately
   if (options().prop.preRegisterMode == options::PreRegisterMode::EAGER)
   {
     Trace("prereg") << "preregister (eager): " << n << std::endl;
     d_theoryEngine->preRegister(n);
+    if (cache)
+    {
+      // cache for registration
+      d_sat_literals.emplace_back(n, d_env.getContext()->getLevel());
+    }
+  }
+}
+
+void TheoryPreregistrar::notifyBacktrack(uint32_t nlevels)
+{
+  (void)nlevels;
```

Comment:
What is the purpose of this line?

Reply:
To silence compiler warnings of unused variables.

Reply:
Note that this argument was introduces in compliance with the new SAT CDCL(T) interface.

Reply:
Yep, makes sense.